scripts/remote_deployment/deploy.sh

```diff
@@ -0,0 +1,228 @@
+  wallet_addresses+=($addr)
+  echo "Wallet $i address: $addr"
+done
+
+default_wallet_address=${wallet_addresses[0]}
+echo "Default wallet address: $default_wallet_address"
+
+# Step 4: Create a subnet
+#echo "$PREFIX Creating a child subnet..."
+#create_subnet_output=$($IPC_CLI subnet create --parent /r314159 --min-validators 3 --min-validator-stake 1 --bottomup-check-period 30 --from $default_wallet_address --permission-mode 0 --supply-source-kind 0 2>&1)
+#echo $create_subnet_output
+#subnet_id=$(echo $create_subnet_output | sed 's/.*with id: \([^ ]*\).*/\1/')
+#
+#echo "Created subnet ID: $subnet_id"
+
+# Step 4 (alternative): Use an already-created subnet
+subnet_id=/r314159/t410fqmlmt6usaeewvxdj3slk6t57ti776ycqsjp2lsa
+
+# Step 5: Generate pubkeys from addresses
+echo "$PREFIX Generating pubkey for wallet addresses... $default_wallet_address"
+for i in {0..2}
+do
+  pubkey=$($IPC_CLI wallet pub-key --wallet-type evm --address ${wallet_addresses[i]})
+  echo "Wallet $i address's pubkey: $pubkey"
+  address_pubkeys+=($pubkey)
+done
+
+# Step 6: Join subnet for addresses in wallet
+#echo "$PREFIX Join subnet for addresses in wallet..."
+#for i in {0..2}
+#do
+#  echo "Joining subnet ${subnet_id} for address ${wallet_address[i]}"
+#  $IPC_CLI subnet join --from ${wallet_addresses[i]} --subnet $subnet_id --public-key ${address_pubkeys[i]} --initial-balance 1 --collateral 10
+#done
+
+# Step 6 (alternative): Assume we already let our addresses join in the subnet
+
+# Step 7: Start validators
+# Step 7.1: Export validator private keys into files
+for i in {0..2}
+do
+  $IPC_CLI wallet export --wallet-type evm --address ${wallet_addresses[i]} --hex > ${IPC_CONFIG_FOLDER}/validator_${i}.sk
+  echo "Export private key for ${wallet_addresses[i]} to ${IPC_CONFIG_FOLDER}/validator_${i}.sk"
+done
+
+# Step 7.2 (optional): Rebuild fendermint docker
+# cd ${IPC_FOLDER}/fendermint
+# make docker-build
+
+# Step 7.3: Read parent net gateway address and registry address
+echo "$PREFIX Reading parent gateway and registry address"
+parent_gateway_address=$(toml get ${IPC_CONFIG_FOLDER}/config.toml subnets[0].config.gateway_addr | tr -d '"')
+parent_registry_address=$(toml get ${IPC_CONFIG_FOLDER}/config.toml subnets[0].config.registry_addr | tr -d '"')
+
+# Step 7.4: Start the bootstrap validator node
+echo "$PREFIX Start the bootstrap validator node"
+cd ${IPC_FOLDER}
+bootstrap_output=$(cargo make --makefile infra/fendermint/Makefile.toml \
+        -e NODE_NAME=validator-1 \
+        -e PRIVATE_KEY_PATH=${IPC_CONFIG_FOLDER}/validator_1.sk \
+        -e SUBNET_ID=${subnet_id} \
+        -e CMT_P2P_HOST_PORT=${CMT_P2P_HOST_PORTS[0]} \
+        -e CMT_RPC_HOST_PORT=${CMT_RPC_HOST_PORTS[0]} \
+        -e ETHAPI_HOST_PORT=${ETHAPI_HOST_PORTS[0]} \
+        -e RESOLVER_HOST_PORT=${RESOLVER_HOST_PORTS[0]} \
+        -e PARENT_REGISTRY=${parent_registry_address} \
+        -e PARENT_GATEWAY=${parent_gateway_address} \
+        -e FM_PULL_SKIP=1 \
```

Comment:
Why use pull skip? This means that you're not pulling and you expect the image to be present in the machine. The side effect is that we will always deploy the testnet with the same image over and over again.

I suggest removing this here, but keeping it below, so you're pulling once per run, and skipping pulls subsequently.

Reply:
Yes got it. Fixed it.